code/game/objects/items/melee/baton.dm

```diff
@@ -0,0 +1,656 @@
+			log_mapping("[src] at [AREACOORD(src)] had an invalid preload_cell_type: [preload_cell_type].")
+		else
+			cell = new preload_cell_type(src)
+	RegisterSignal(src, COMSIG_PARENT_ATTACKBY, .proc/convert)
+	update_appearance()
+
+/obj/item/melee/baton/security/get_cell()
+	return cell
+
+/obj/item/melee/baton/security/suicide_act(mob/user)
+	if(cell?.charge && active)
+		user.visible_message(span_suicide("[user] is putting the live [name] in [user.p_their()] mouth! It looks like [user.p_theyre()] trying to commit suicide!"))
+		. = (FIRELOSS)
+		attack(user,user)
+	else
+		user.visible_message(span_suicide("[user] is shoving the [name] down their throat! It looks like [user.p_theyre()] trying to commit suicide!"))
+		. = (OXYLOSS)
+
+/obj/item/melee/baton/security/Destroy()
+	if(cell)
+		QDEL_NULL(cell)
+	UnregisterSignal(src, COMSIG_PARENT_ATTACKBY)
+	return ..()
+
+/obj/item/melee/baton/security/proc/convert(datum/source, obj/item/I, mob/user)
+	SIGNAL_HANDLER
+
+	if(istype(I,/obj/item/conversion_kit) && convertible)
+		var/turf/source_turf = get_turf(src)
+		var/obj/item/melee/baton/baton = new (source_turf)
+		baton.alpha = 20
+		playsound(source_turf, 'sound/items/drill_use.ogg', 80, TRUE, -1)
+		animate(src, alpha = 0, time = 10)
+		animate(baton, alpha = 255, time = 10)
+		qdel(I)
+		qdel(src)
+
+/obj/item/melee/baton/security/Exited(atom/movable/mov_content)
+	. = ..()
+	if(mov_content == cell)
+		cell.update_appearance()
+		cell = null
+		active = FALSE
+		update_appearance()
+
+/obj/item/melee/baton/security/update_icon_state()
+	if(active)
+		icon_state = "[initial(icon_state)]_active"
+		return ..()
+	if(!cell)
+		icon_state = "[initial(icon_state)]_nocell"
+		return ..()
+	icon_state = "[initial(icon_state)]"
+	return ..()
+
+/obj/item/melee/baton/security/examine(mob/user)
+	. = ..()
+	if(cell)
+		. += span_notice("\The [src] is [round(cell.percent())]% charged.")
+	else
+		. += span_warning("\The [src] does not have a power source installed.")
+
+/obj/item/melee/baton/security/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = W
+		if(cell)
+			to_chat(user, span_warning("[src] already has a cell!"))
+		else
+			if(C.maxcharge < cell_hit_cost)
+				to_chat(user, span_notice("[src] requires a higher capacity cell."))
+				return
+			if(!user.transferItemToLoc(W, src))
+				return
+			cell = W
+			to_chat(user, span_notice("You install a cell in [src]."))
+			update_appearance()
+
+	else if(W.tool_behaviour == TOOL_SCREWDRIVER)
+		tryremovecell(user)
+	else
+		return ..()
+
+/obj/item/melee/baton/security/proc/tryremovecell(mob/user)
+	if(cell && can_remove_cell)
+		cell.forceMove(drop_location())
+		to_chat(user, span_notice("You remove the cell from [src]."))
+
+/obj/item/melee/baton/security/attack_self(mob/user)
+	if(cell?.charge >= cell_hit_cost)
+		active = !active
+		balloon_alert(user, "[src] is now [active ? "on" : "off"].")
```

Comment:
typically, balloon alerts are concise / don't have punctuation (something like, "baton active") (though mothblocks might not prefer balloon alerts for security batons)

Reply:
telescopic batons already use balloon alerts for that (I'm not using the transforming component here, the force, attack verbs and weight class of stunbatons are unchanged when toggled on/off).

Reply:
I already told Melbert that the balloon alerts for transforming batons is already really bulky.

```suggestion
		balloon_alert(user, "turned [active ? "on" : "off"]")
```